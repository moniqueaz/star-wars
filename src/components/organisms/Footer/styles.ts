import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${( { theme } ) => css`
    text-align: center;
    background-color: ${theme.colors.black};
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    height: ${theme.height.footer.mobile};
    @media(min-width: 1024px){
      height: ${theme.height.footer.desk};
    }
  `}
`;
