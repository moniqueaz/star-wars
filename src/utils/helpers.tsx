export const fetcher = (url: string) => fetch(url).then(res => res.json());

export const getRandomInteger = (min: number, max: number) => {
  return Math.floor(Math.random() * (max - min)) + min;
};
