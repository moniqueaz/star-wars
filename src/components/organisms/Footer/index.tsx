import * as S from './styles';
import Text from 'components/atoms/Text';
import { Container } from 'components/atoms/Container';

const Footer = () => (
  <S.Wrapper>
    <Container>
      <Text light>Star Wars Card Game</Text>
    </Container>
  </S.Wrapper>
);

export default Footer;
