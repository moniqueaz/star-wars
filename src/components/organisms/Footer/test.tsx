import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Footer from '.';

type ContainerProps = {
  children: React.ReactNode
}

jest.mock('components/atoms/Text', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Text"></div>;
    },
  };
} );

jest.mock('components/atoms/Container', () => {
  return {
    Container: function Mock( { children }: ContainerProps) {
      return <div data-testid="Mock Container">{children}</div>;
    },
  };
} );

describe('<Footer />', () => {
  it('should render Container', () => {
    const { container } = renderWithTheme(<Footer />);
    expect(screen.getByTestId('Mock Container')).toBeInTheDocument();
    expect(screen.getByTestId('Mock Text')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  } );
} );
