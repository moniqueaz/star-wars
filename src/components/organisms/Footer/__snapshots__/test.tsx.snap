// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render Container 1`] = `
.c0 {
  text-align: center;
  background-color: #333;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 4.6rem;
}

@media (min-width:1024px) {
  .c0 {
    height: 7.6rem;
  }
}

<div
  class="c0"
>
  <div
    data-testid="Mock Container"
  >
    <div
      data-testid="Mock Text"
    />
  </div>
</div>
`;
